assets/runtime/functions

```diff
@@ -748,10 +748,18 @@ gitlab_configure_secrets() {
     return 1
   fi
 
+  if [[ -z $GITLAB_SECRETS_ENCRYPTED_SETTINGS_KEY_BASE ]]; then
+    echo "ERROR: "
+    echo "  Please configure the GITLAB_SECRETS_ENCRYPTED_SETTINGS_KEY_BASE parameter."
+    echo "  Cannot continue. Aborting..."
+    return 1
+  fi
+
```

Comment:
Making this optional might introduce an issue when generating the `secrets.yml`. Since it is referenced in the template, the resulting secrets.yml should look like :

```yml
production:
  # db_key_base is used to encrypt for Variables. Ensure that you don't lose it.
  # If you change or lose this key you will be unable to access variables stored in database.
  # Make sure the secret is at least 30 characters and all random,
  # no regular words or you'll be exposed to dictionary attacks.
  db_key_base: long-and-random-alphanumeric-string
  secret_key_base: long-and-random-alphanumeric-string
  otp_key_base: long-and-random-alphanumeric-string
  encrypted_settings_key_base: 

development:
  db_key_base: development

test:
  db_key_base: test
```

And startup might be blocked. I'm re-building the image to test it and check.

Reply:
These secret will be [handled using `.present?`](https://gitlab.com/gitlab-org/gitlab/-/blob/eeaeeea1af5a97285977120c87188da3e60b90f6/config/initializers/01_secret_token.rb#L67-72). [This method](https://api.rubyonrails.org/classes/Object.html#method-i-present-3F) returns true only when non-blank value is set, so I believe we can leave it blank (random secret might be generated). I look forward to your report.

Reply:
Totally agree on this point, but I'm afraid of the yaml syntax of the generated `secrets.yml` that might not be considered as valid for the yaml parser because of the empty value.

Reply:
Ok I just tested it with a local build, and this works fine, I pushed the related changeset.